dali/pipeline/data/tensor_list.cc

```diff
@@ -1121,6 +1126,8 @@ void TensorList<Backend>::UpdatePropertiesFromSamples(bool contiguous) {
     DALI_ENFORCE(GetLayout() == tensors_[i].GetLayout(),
                  make_string("Samples must have the same layout, expected: ", GetLayout(),
                              " got: ", tensors_[i].GetLayout(), " at ", i, "."));
+    // if tensor is empty, its pinned status, order and device_id is rather irrelevant
+    if (tensors_[i].nbytes() == 0) continue;
```

Comment:
And we just kept checking all the samples after such propagation.

Reply:
Done